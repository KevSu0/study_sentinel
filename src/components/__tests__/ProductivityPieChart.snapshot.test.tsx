/** @jest-environment jsdom */
import 'fake-indexeddb/auto';
import React from 'react';
import { render } from '@testing-library/react';
import ProductivityPieChart from '@/components/dashboard/productivity-pie-chart';

describe('ProductivityPieChart snapshot (structure only)', () => {
  it('matches structure snapshot for typical data', () => {
    const data = [
      { name: 'Task: Focus', productiveDuration: 25 * 60, pausedDuration: 5 * 60, pauseCount: 1, focusPercentage: 83 },
      { name: 'Routine: Reading', productiveDuration: 15 * 60, pausedDuration: 0, pauseCount: 0, focusPercentage: 100 },
    ];
    const { asFragment } = render(<ProductivityPieChart data={data} focusScore={90} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

