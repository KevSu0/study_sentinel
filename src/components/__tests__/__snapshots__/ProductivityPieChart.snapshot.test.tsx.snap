// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductivityPieChart snapshot (structure only) matches structure snapshot for typical data 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm h-full min-h-[380px] flex flex-col"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-lg font-semibold leading-none tracking-tight"
      >
        Today's Productivity
      </div>
      <div
        class="text-sm text-muted-foreground"
      >
        Hover over a slice to see detailed info.
      </div>
    </div>
    <div
      class="flex-grow flex items-center justify-center p-0 pb-2 relative"
    >
      <div
        class="absolute inset-0 flex flex-col items-center justify-center pointer-events-none text-center"
      >
        <p
          class="text-4xl font-bold tracking-tighter"
        >
          40m
        </p>
        <p
          class="text-sm text-muted-foreground"
        >
          Productive Time
        </p>
        <p
          class="text-2xl font-bold text-green-500 mt-2"
        >
          90% 
          <span
            class="text-sm font-medium text-muted-foreground"
          >
            Focus
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
